standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HiveMetaStore.java

```diff
@@ -9322,9 +9334,25 @@ public void processContext(ServerContext serverContext, TTransport tTransport, T
     if (startLock != null) {
       signalOtherThreadsToStart(tServer, startLock, startCondition, startedServing);
     }
+
+    // If dynamic service discovery through ZooKeeper is enabled, add this server to the ZooKeeper.
+    if (MetastoreConf.getVar(conf, ConfVars.THRIFT_SERVICE_DISCOVERY_MODE)
+            .equalsIgnoreCase("zookeeper")) {
+      zooKeeperHelper = MetastoreConf.getZKConfig(conf);
+      String serverInstanceURI = getServerInstanceURI(port);
+      zooKeeperHelper.addServerInstanceToZooKeeper(serverInstanceURI, serverInstanceURI, null,
```

Comment:
do we need the same value for path and data also ? cant we just use one ?

Reply:
For HiveServer2 we use different values for the path and the data, so addServerInstanceToZooKeeper accepts both. For HiveMetaStore, we don't need data to be different from the path, so we pass same value for path and data. We can pass only path to addServerInstanceToZooKeeper() and leave data to be null and then the function can use the same value for the data as the path for HiveMetaStore. But passing both of those explicitly leaves room in case we pass different path and data for HiveMetaStore as well. In future, we may want to publish HiveMetaStore configs in ZooKeeper just like HiveServer2. So I chose the API to be extensible rather that adding few lines of code dealing with null data in addServerInstanceToZooKeeper().